### PR TITLE
make SqlSaga great again

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@Particular/sqlpersistence-maintainers review please :)

File: src/ScriptBuilder/Saga/SagaDefinitionReader.cs [hunk 1]
```diff
@@ -42,29 +46,53 @@ public static bool TryGetSqlSagaDefinition(TypeDefinition type, out SagaDefiniti
         return true;
     }
 
+    static string GetCorrelationPropertyName(TypeDefinition type)
+    {
+        var instructions = type.Properties.Single(_ => _.Name == "CorrelationPropertyName").GetMethod.Body.Instructions;
+        if (instructions.Count == 0)
```
Comment (SzymonPobiega) [inline src/ScriptBuilder/Saga/SagaDefinitionReader.cs]: When does this happen?

File: src/ScriptBuilder/Saga/SagaDefinitionReader.cs [hunk 2]
```diff
@@ -42,29 +46,53 @@ public static bool TryGetSqlSagaDefinition(TypeDefinition type, out SagaDefiniti
...
-            return;
-        }
-        var baseTypeFullName = type.BaseType.FullName;
-        if (baseTypeFullName.StartsWith("NServiceBus.Persistence.Sql.SqlSaga"))
+        if (type.BaseType == null)
         {
-            throw new ErrorsException($"The type '{type.FullName}' inherits from NServiceBus.Persistence.Sql.SqlSaga but does not contain a [CorrelatedSagaAttribute] or [AlwaysStartNewSagaAttribute].");
+            return false;
         }
+        return type.BaseType.FullName.StartsWith("NServiceBus.Persistence.Sql.SqlSaga");
```
Comment (SzymonPobiega) [inline src/ScriptBuilder/Saga/SagaDefinitionReader.cs]: Do we care to check if the base type is defined in the correct assembly?

File: src/SqlPersistence/Saga/IMessagePropertyMapper.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    using System;
+    using System.Linq.Expressions;
+
+    public interface IMessagePropertyMapper
+    {
+        /// <summary>
+        /// Configures the mapping between <see cref="SqlSaga{TSagaData}.CorrelationPropertyName"/> and <typeparamref name="TMessage" />.
+        /// </summary>
+        /// <typeparam name="TMessage">The message type to map to.</typeparam>
+        /// <param name="messageProperty">An <see cref="Expression{TDelegate}" /> that represents the message.</param>
```
Comment (SzymonPobiega) [inline src/SqlPersistence/Saga/IMessagePropertyMapper.cs]: that represents the message *property* ?

File: src/ScriptBuilder/Saga/SagaDefinitionReader.cs (hunk 1 above)
Comment (SimonCropp) [inline src/ScriptBuilder/Saga/SagaDefinitionReader.cs]: good point. will remove

**Comment by SimonCropp** [comment]
@SzymonPobiega addressed comments

**Pull request merged**